src/java/htsjdk/tribble/util/ParsingUtils.java

```diff
@@ -475,4 +475,15 @@ public static String appendToPath(String filepath, String indexExtension) {
         }
         return tabxIndex;
     }
+
+    /**
+     * Test if path ends with extension after stripping URL query string
+     */
+    public static boolean pathHasExtension(final String path, final String extension) {
+        final String strippedPath = (path.startsWith("http://") || path.startsWith("https://")) ?
```

Comment:
Should this method work with arbitrary protocols, rather than just `http`/`https`? You might want to rely on the built-in parsing available in java's `URI` class here.